x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/categorization_partition_field/categorization_per_partition_input.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useCallback, useContext, useMemo } from 'react';
+import { EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+
+import { JobCreatorContext } from '../../../job_creator_context';
+import { Field } from '../../../../../../../../../common/types/fields';
+import { createFieldOptions } from '../../../../../common/job_creator/util/general';
+
+interface Props {
+  fields: Field[];
+  changeHandler(i: string | null): void;
+  selectedField: string | null;
+}
+
+export const CategorizationPerPartitionFieldSelect: FC<Props> = ({
+  fields,
+  changeHandler,
+  selectedField,
+}) => {
+  const { jobCreator } = useContext(JobCreatorContext);
+  const options: EuiComboBoxOptionOption[] = useMemo(
+    () => [...createFieldOptions(fields, jobCreator.additionalFields)],
+    [fields, jobCreator]
+  );
+
+  const selection: EuiComboBoxOptionOption[] = useMemo(() => {
+    const selectedOptions: EuiComboBoxOptionOption[] = [];
+    if (selectedField !== null) {
+      selectedOptions.push({ label: selectedField });
+    }
+    return selectedOptions;
+  }, [selectedField]);
+
+  const onChange = useCallback(
+    (selectedOptions: EuiComboBoxOptionOption[]) => {
+      const option = selectedOptions[0];
+      if (typeof option !== 'undefined') {
+        changeHandler(option.label);
+      } else {
+        changeHandler(null);
+      }
+    },
+    [changeHandler]
```

Comment:
is this correct? `changeHandler` is a reference to a function. it looks like this should be watching `selectedOptions[0]`
it could also be simplified to `changeHandler(selectedOptions[0] ?? null)`

Reply:
Updated https://github.com/elastic/kibana/pull/75061/commits/4b2f2e01b24ae30e0bd8a61ddf877015e5785ecb